Sources/ScipioS3Storage/Compressor.swift

```diff
@@ -0,0 +1,81 @@
+import Foundation
+import System
+import Compression
+import AppleArchive
+
+struct Compressor {
+    private let fileManager: FileManager = .default
+
+    init() {
+        try? fileManager.createDirectory(at: temporaryDirectory, withIntermediateDirectories: true)
+    }
+
+    func compress(_ directoryPath: URL) throws -> Data {
+        guard let keySet else { throw Error.initializationError }
+
+        let source = FilePath(directoryPath.path)
+
+        let archivePath = temporaryDirectory.appendingPathComponent("\(UUID().uuidString).aar")
+        defer { try? fileManager.removeItem(at: archivePath) }
+
+        try ArchiveByteStream.withFileStream(
+            path: FilePath(archivePath.path),
+            mode: .writeOnly,
+            options: [.create, .truncate],
+            permissions: [.ownerReadWrite, .groupRead, .otherRead]
+        ) { file in
+            try ArchiveByteStream.withCompressionStream(using: .lzfse, writingTo: file) { compressor in
+                try ArchiveStream.withEncodeStream(writingTo: compressor) { encoder in
+                    try encoder.writeDirectoryContents(archiveFrom: source, keySet: keySet)
+                }
+            }
+        }
+
+        guard let data = fileManager.contents(atPath: archivePath.path) else {
+            throw Error.compressionError
+        }
+
+        return data
+    }
+
+    func extract(_ archiveData: Data, to destinationPath: URL) throws {
+        let destination = FilePath(destinationPath.path)
+
+        let temporaryPath = temporaryDirectory.appendingPathComponent("\(UUID().uuidString).aar")
+        fileManager.createFile(atPath: temporaryPath.path, contents: archiveData)
+        defer { try? fileManager.removeItem(at: temporaryPath) }
+
+        try fileManager.createDirectory(at: destinationPath, withIntermediateDirectories: true)
+
+        _ = try ArchiveByteStream.withFileStream(
```

Comment:
Isn't its return value discardable value?

Reply:
This returns a return value of closure.
https://developer.apple.com/documentation/applearchive/archivebytestream/3588517-withfilestream

It is not needed here because closure writes this to the file.